### ISSUE TITLE
Away status of a channel gets cleared on /names

### ISSUE THREAD

**Issue by wowaname** [weechat/weechat#434]
I'm using the away-notify cap when encountering this issue; I have not tried normal away checking with the WHO command every x minutes but I believe it would produce a similar issue.

I have the nicklist enabled for channels, and away users grey out as is expected with my configuration, but when I issue a /names for the channel, these away users reset and appear as unaway (not greyed out). When I /who the channel, the away statuses return to normal, so it is not a dire issue, but I believe it would still be sane to expect that away nicks stay away.

I know that a /names actually refreshes the nicklist, so on an improperly configured or desynced server, a /names will remove/add nicks from the nicklist if it sees that the /names output doesn't match that of the nicklist. Still, for normal use I would think that resetting away status for everyone in a channel would be something a user wouldn't expect.

Also -- this may qualify as a new issue, but it relates to away status -- it would be nice if the /names output showed away nicks as greyed out as well.

**Comment by flashcode** [comment]
The `/names` command does a full resync of nicks with the channel, ie the nicks in WeeChat are destroyed (so everything is cleared, including the away status).
I marked this issue as a bug, but I consider the current behavior as the expected one, just that it could be improved by saving the away status (and maybe other things) and put it back on nick after resync.

**Comment by wowaname** [comment]
Yeah, as I have said, I realise that /nicks rewrites the nicklist but I just see it being more intuitive to most people to keep the away nicks instead of wiping them.
